desc_source.go

```diff
@@ -206,8 +206,33 @@ func DescriptorSourceFromServer(_ context.Context, refClient *grpcreflect.Client
 	return serverSource{client: refClient}
 }
 
+// DescriptorSourceFromServerWithAdditionalProtosets creates a DescriptorSource
+// that uses the given gRPC reflection client to interrogate a server for descriptor
+// information and the supplied protosets as a source of additional symbols.
+func DescriptorSourceFromServerWithAdditionalProtosets(_ context.Context, refClient *grpcreflect.Client, fileNames ...string) DescriptorSource {
```

Comment:
Sorry I didn't leave this comment in earlier iteration, before you re-worked the flags. But I don't think we want to widen the API like this. Instead, I think you could have a custom implementation of `DescriptorSource` in the `main` package that just had two underlying sources -- reflection source, then the proto or protoset source.

Its implementation would just look at the reflection source for listing services, but could then use both when resolving symbols (consult first then fallback to other on error).